src/main/webapp/app/exam/examExercisesNavigationService.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({ providedIn: 'root' })
+export class ExamExercisesNavigationService {
```

Comment:
Why do you need a separate service here? Can't this be merged with the `LiveExamExerciseUpdateService`? It seems to be strongly coupled to the use case of live-exam-updates as well...

Reply:
I also felt like these two could be merged, but I was not sure about my logic. My workflow with behaviour subjects seems to be working and has not been rejected (yet), therefore I now refactored it and merged both services.

Thank you. Now it is less confusing. I simply check if the `problemStatement` is `''` or not and navigate or update the problem statement thereafter.